src/components/Combobox/Dynamic/DynamicCombobox.tsx

```diff
@@ -129,16 +135,16 @@ const DynamicComboboxInner = <T extends Option>(
           {startAdornment && typed && <Box>{startAdornment(value)}</Box>}
 
           <Box
-            id={id}
             as="input"
             type="text"
             className={classNames(inputRecipe({ size, error }))}
             disabled={disabled}
             {...props}
-            {...getInputProps({
-              id,
-              ref,
-            })}
+            {...inputProps}
+            // There is mismatch between desert box onChange type and downshift on change event
+            onChange={
+              inputProps.onChange as unknown as FormEventHandler<HTMLElement>
```

Comment:
wrapping this into function would not help? 
```
onChange={(event) => inputProps.onChange(event)}
```

or add your handler? Because that force-casting can be dangerous, i would recommend to be more explicit